staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -187,7 +187,21 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	}
 	defaultVersionedObject := indirectArbitraryPointer(versionedPtr)
 	kind := fqKindToRegister.Kind
-	hasSubresource := len(subresource) > 0
+	isSubresource := len(subresource) > 0
+
+	// If there is a subresource, namespace scoping is defined there
+	namespaceScoped := true
+	if isSubresource {
+		parentStorage, ok := a.group.Storage[resource]
+		if !ok {
+			return nil, fmt.Errorf("missing parent storage: %q", resource)
+		}
+		if scoper, ok := parentStorage.(rest.Scoper); ok {
+			namespaceScoped = scoper.NamespaceScoped()
+		}
+	} else if scoper, ok := storage.(rest.Scoper); ok {
```

Comment:
defaulting is dangerous... we should require the storage to implement this.

Reply:
> defaulting is dangerous... we should require the storage to implement this.

I will point out that you'll notice very quickly. Now that I found parents, this may not be so bad.  I double check.